lib/policies/proxy/proxy.js

```diff
@@ -51,15 +54,49 @@ module.exports = function (params, config) {
   if (endpoint.url) {
     return function proxyHandler (req, res) {
       logger.debug(`proxying to ${endpoint.url}, ${req.method} ${req.url}`);
-      proxy.web(req, res, { target: endpoint.url });
+      proxy.web(req, res, proxyOptions);
     };
   }
 
   // multiple urls will load balance, defaulting to round-robin
   const strategy = params.strategy || 'round-robin';
-  const balancer = createStrategy(strategy, proxy, endpoint.urls);
+  const balancer = createStrategy(strategy, proxy, proxyOptions, endpoint.urls);
 
   return function balancerProxyHandler (req, res) {
     return balancer.proxy(req, res);
   };
 };
+
+// Parse endpoint URL if single URL is provided.
+// Extend proxy options by allowing and parsing keyFile and certFile.
+function parseProxyOptions (proxyOptions = {}, endpoint = {}) {
+  let parsedProxyOptions = {};
+
+  if (Object.keys(endpoint).includes('proxyOptions')) {
```

Comment:
We don't need to enumerate the whole `endpoint` keys. We can just check for property existence:

`if (endpoint.proxyOptions) {…}`

Reply:
I didn't realise this enumerates all keys. Assumed (bad!) it works similarly to .includes? in ruby. Fixing.

Reply:
I would assume that `.includes` does the same thing as Javascript.

My point here is that instead of enumerating all the object keys and then check a property presence from an array of strings, we can check the property existence directly.

```diff
- if (Object.keys(endpoint).includes('proxyOptions')) {
+ if (endpoint.proxyOptions) {
```

https://runkit.com/59ef79493b34ed00123dd9b0/59faf9e0a81fc60012e4f649

Reply:
I get your point, and have made the changes. Just not pushed yet. Waiting on responses to my other comments. 

Also, I meant .key? not includes?! key? does not iterate.